core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -713,29 +725,58 @@ wasm_destroy_memory(WASMMemoryInstance *memory)
 static void
 tables_deinstantiate(WASMModuleInstance *module_inst)
 {
-    if (module_inst->tables) {
-        wasm_runtime_free(module_inst->tables);
+    if (!module_inst) {
+        return;
     }
-#if WASM_ENABLE_MULTI_MODULE != 0
+
+#if WASM_ENABLE_MULTI_MODULE == 0
+    if (!module_inst->tables) {
+        return;
+    }
+
+    WASMModule *module = module_inst->module;
+    /* only imported tables */
+    for (uint32 i = 0; i < module->import_table_count; i++) {
+        WASMTableInstance *table = module_inst->tables[i];
+        if (!table) {
+            continue;
+        }
+
+        table_elem_type_t *table_elems =
+            wasm_locate_table_elems(module, table, i);
```

Comment:
Same as aot_runtime.c, should we check whether `table_elems` is NULL, and free it only when it is not NULL?